寻找“水仙花数”。

In [36]:
for i in range(100,1000):
    a=int(i/100)
    b=int((i-int(i/100)*100)/10)
    c=i-int(i/100)*100
    d=int(c-int(c/10)*10)
    if a**3+b**3+d**3==i:
        print(i)
        

153
370
371
407


"""
找出100~999之间的所有水仙花数
水仙花数是各位立方和等于这个数本身的数
如: 153 = 1**3 + 5**3 + 3**3
Version: 0.1
Author: 骆昊
Date: 2018-03-02
"""

for num in range(100, 1000):
    low = num % 10                 # %求余数...
    mid = num // 10 % 10
    high = num // 100
    if num == low ** 3 + mid ** 3 + high ** 3:
        print(num)                 #好简单啊...

In [42]:
78%10

8

In [ ]:
寻找“完美数”。

In [50]:
import math
for i in range(1000):
    a=0
    for n in range(1,i):
        if i%n==0:
            a +=n
            if a==i:
                print(i) #为什么还多一个24呢？
                        
            

6
24
28
496


In [ ]:
"""
找出1~9999之间的所有完美数
完美数是除自身外其他所有因子的和正好等于这个数本身的数
例如: 6 = 1 + 2 + 3, 28 = 1 + 2 + 4 + 7 + 14
Version: 0.1
Author: 骆昊
Date: 2018-03-02
"""
import time
import math

start = time.clock()
for num in range(1, 10000):
    sum = 0
    for factor in range(1, int(math.sqrt(num)) + 1):
        if num % factor == 0:
            sum += factor
            if factor > 1 and num / factor != factor: #差了这句话
                sum += num / factor
    if sum == num:
        print(num)
end = time.clock()
print("执行时间:", (end - start), "秒")

In [45]:
import time
help(time.clock)

Help on built-in function clock in module time:

clock(...)
    clock() -> floating point number
    
    Return the CPU time or real time since the start of the process or since
    the first call to clock().  This has as much precision as the system
    records.



“百钱百鸡”问题。

In [35]:
for a in range(100):
    if a%5==0:
        for b in range(100):
            if b%3==0:
                for c in range(100):
                    m=a+b+c
                    n=a/5+b/3+c*3
                    if m==100 and n==100:
                        print("公鸡、母鸡、小鸡的数量是",a/5,b/3,c)

公鸡、母鸡、小鸡的数量是 0.0 25.0 25
公鸡、母鸡、小鸡的数量是 4.0 18.0 26
公鸡、母鸡、小鸡的数量是 8.0 11.0 27
公鸡、母鸡、小鸡的数量是 12.0 4.0 28


"""
求解《百钱百鸡》问题
1只公鸡5元 1只母鸡3元 3只小鸡1元 用100元买100只鸡
问公鸡 母鸡 小鸡各有多少只
Version: 0.1
Author: 骆昊
Date: 2018-03-02
"""

for x in range(0, 20): #这边他设的x为公鸡的数量，5钱一只，100钱最多能买20只，所以取区间（0，20）
    for y in range(0, 33):
        z = 100 - x - y
        if 5 * x + 3 * y + z / 3 == 100:
            print('公鸡: %d只, 母鸡: %d只, 小鸡: %d只' % (x, y, z))

# 要理解程序背后的算法 - 穷举法

生成“斐波拉切数列”。

In [34]:
a=0
b=1
i=0
while i<13:
    a,b=b,a+b
    i+=1
    print(a,end=',')

1,1,2,3,5,8,13,21,34,55,89,144,233,

"""
输出斐波那契数列的前20个数
1 1 2 3 5 8 13 21 ...
Version: 0.1
Author: 骆昊
Date: 2018-03-02
"""

a = 0
b = 1
for _ in range(20):  #这边的_意味着你不想用这个值，你只是需要它的循环，忽略它
    (a, b) = (b, a + b)
    print(a, end=' ')    #比我写的简洁啊

In [ ]:
import os
import time
def main():
    content = '豪绅🐮🍺…………豪绅🐮🍺.......'
    while True:
        # 清理屏幕上的输出
        os.system('cls')  # os.system('clear')
        print(content)
        # 休眠200毫秒
        time.sleep(0.5)
        content = content[1:] + content[0]
        
if __name__ == '__main__':
    main()